lib/logstash/inputs/tcp.rb

```diff
@@ -177,15 +180,20 @@ def register
     validate_ssl_config!
 
     if server?
-      @loop = InputLoop.new(@id, @host, @port, DecoderImpl.new(@codec, self), @tcp_keep_alive, java_ssl_context)
+      @port_reservation = port_management.reserve(addr: @host, port: @port) do |reserved_addr, reserved_port|
+        @loop = InputLoop.new(@id, reserved_addr, reserved_port, DecoderImpl.new(@codec, self), @tcp_keep_alive, java_ssl_context)
+      end
```

Comment:
I don't think we need to use blocks here since we're not wrapping behavior:

```suggestion
      port_management.reserve(port: @port) # if this succeeds, there is a reservation for the port
      @loop = InputLoop.new(@id, @host, @port, DecoderImpl.new(@codec, self), @tcp_keep_alive, java_ssl_context)
```

Also we should set the reservation scope for the port alone. Not sure if it's worth differentiating the addr, we can be conservative here and allow the port to be reserved regardless of the addr.